dot/state/service.go

```diff
@@ -51,22 +51,28 @@ type Service struct {
 	// Below are for testing only.
 	BabeThresholdNumerator   uint64
 	BabeThresholdDenominator uint64
+
+	// Below are for state trie online pruner
+	retainBlocks int64
+	gcMode       string
 }
 
 // NewService create a new instance of Service
-func NewService(path string, lvl log.Lvl) *Service {
+func NewService(path string, lvl log.Lvl, gcMode string, retainBlocks int64) *Service {
```

Comment:
maybe change `gcMode` to its own type instead of just `string`?

Reply:
Done